### PR TITLE
[ff-2388] get_bandit_actions: supply actions as a dict instead of list

### PR DESCRIPTION
**Pull request by schmit**
[//]: #  (Link to the issue corresponding to this chunk of work)
Fixes: #ff-2388

## Motivation and Context
[//]: #  (Why is this change required? What problem does it solve?)

Currently actions are supplied as a list, which does not guarantee that the action keys are unique. To ensure uniqueness, require a dictionary mapping actions to their contexts instead

## Description
[//]: # (Describe your changes in detail)

- Change interface to use Dict instead of List
- Update `bandit.py` to use dicts for computation instead of lists
- Update tests to supply Dicts instead of Lists
- Minor version bump

## How has this been tested?
[//]: # (Please describe in detail how you tested your changes)

All tests pass

[//]: # (OPTIONAL)
[//]: # (Add one or multiple labels: enhancement, refactoring, bugfix)

### CODE REVIEW

File: eppo_client/bandit.py [hunk 1]
```diff
@@ -108,101 +71,85 @@ def evaluate_bandit(
...
-        selected_idx = next(
-            idx
-            for idx, action_context in enumerate(actions_with_contexts)
-            if action_context.action_key == selected_action
-        )
-
-        optimality_gap = (
-            max(score for _, score in action_scores) - action_scores[selected_idx][1]
-        )
+        optimality_gap = max(action_scores.values()) - action_scores[selected_action]
```
Comment (giorgiomartini0) [inline eppo_client/bandit.py:91]: Sweet bonus refactor!

**Review by giorgiomartini0 — APPROVED** [review]

File: eppo_client/bandit.py [hunk 2]
```diff
@@ -104,101 +71,85 @@ def evaluate_bandit(
         flag_key: str,
         subject_key: str,
         subject_attributes: Attributes,
-        actions_with_contexts: List[ActionContext],
+        actions: ActionContexts,
```
Comment (aarsilv) [inline eppo_client/bandit.py:74]: should we still keep this named`actions_with_contexts`?

File: eppo_client/version.py [hunk 3]
```diff
@@ -1 +1 @@
-__version__ = "3.1.4"
+__version__ = "3.2.0"
```
Comment (aarsilv) [inline eppo_client/version.py:1]: _technically_ this interface change is a non-backward compatible breaking change, but fine doing the 3.2 thing

File: test/bandit_test.py [hunk 4]
```diff
@@ -333,3 +319,73 @@ def test_evaluate_bandit():
     assert evaluation.gamma == bandit_model.gamma
     assert evaluation.action_score == 4.0
     assert pytest.approx(evaluation.action_weight, rel=1e-2) == 0.4926
+
+
+def test_bandit_no_action_contexts():
```
Comment (aarsilv) [inline test/bandit_test.py:324]: nice!

File: test/bandit_test.py [hunk 5]
```diff
@@ -333,3 +319,73 @@ def test_evaluate_bandit():
...
+        action_probability_floor=0.1,
+        coefficients=coefficients,
+    )
+
+    evaluator = BanditEvaluator(sharder=DeterministicSharder({}))
+    evaluation = evaluator.evaluate_bandit(
+        flag_key,
+        subject_key,
+        subject_attributes,
+        {"action1": Attributes.empty(), "action2": Attributes.empty()},
```
Comment (aarsilv) [inline test/bandit_test.py:381]: 👍

**Review by aarsilv — APPROVED** [review]
📈 Love it!

File: eppo_client/version.py (hunk 3 above)
Comment (schmit) [reply eppo_client/version.py:1]: Yup my thought exactly

File: eppo_client/bandit.py (hunk 2 above)
Comment (schmit) [reply eppo_client/bandit.py:74]: Thought about it -- given that we expect a dictionary now, I think it's more obvious that we need extra information to go with the key

**Pull request merged**